packages/react/src/components/theme-root/index.tsx

```diff
@@ -19,25 +20,26 @@ type Props = {
 };
 
 /**
- * This component is to be rendered as close as the app root.
- * It sets the proper class names so that child components on all levels can use the theming properties.
+ * This component will apply the necessary class names to the <body> element.
+ * That way the child components on all levels can use the theming properties.
+ * This also helps with any components portaling out of the SlashIDProvider hierarchy.
  */
 export function ThemeRoot({ children, theme = "light", className }: Props) {
-  return (
-    <div
-      className={clsx(
-        "sid-theme-root",
-        `sid-theme-root__${theme}`,
-        themeClass,
-        {
-          [darkTheme]: theme === "dark",
-          [autoTheme]: theme === "auto",
-          [lightThemeVars]: theme === "light",
-        },
-        className
-      )}
-    >
-      {children}
-    </div>
-  );
+  useLayoutEffect(() => {
```

Comment:
@ahmed-gz @jake-slashid what do you think? Tested it locally and it seems to behave as expected.